### DISCUSSION TITLE
Developers, I need help...

### DISCUSSION THREAD

**Discussion by dcorson-ticino-com** [jamulussoftware/jamulus#1215] [Technical Support & Help]
Does anyone have an idea why I see different behaviour between doing File>Exit and clicking the close button X at the top right ?
The close button is apparently not calling the closeEvent which means the ini file is not being written.
Google hasn't surfaced any worthwhile possibilities.
I am stuck...
Don
(Other than this problem NewUI-03 is ready with dockable windows, I like it a lot.)

**Comment by ann0see** [comment]
I'm not sure here, but can you figure it out in the code?

**Reply by dcorson-ticino-com** [reply]
That's the problem, there is something I don't understand happening **in my own code**.
I'm going to forget it for a day or two and try again this coming weekend.  :)

**Reply by ann0see** [reply]
Ok. I hope somebody else looks here. I'm busy fixing another graphical bug at the moment.

**Comment by hoffie** [comment]
Hm, as far as I can see, Ctrl+Q / File->Exit just calls close():
https://github.com/jamulussoftware/jamulus/blob/be821839e51ab00f7504fda4def6f595ed1146d2/src/clientdlg.cpp#L258

This is will apparently fire the close event ([QCloseEvent](https://doc.qt.io/archives/qt-4.8/qcloseevent.html)), which ends up being handled in [closeEvent](https://github.com/jamulussoftware/jamulus/blob/be821839e51ab00f7504fda4def6f595ed1146d2/src/clientdlg.cpp#L590).

I just added a quick `qDebug() << "close event";` there and it triggered both when closing via my window manager and when using the menu.

Can you confirm how this behaves in your case?
What's your environment / operating system?

**Reply by dcorson-ticino-com** [reply]
Thanks Christian,
I am running Win10 and the released code works.  I have added docked windows, to do that I had to use a QMainWindow making it the main window and owner of the title bar and no longer the ClientDlg.  The close triggered in the ClientDlg File>Exit works, the close from the title bar X doesn't.  In the mean time I am sure that I am missing a Signal/Slot between the QMainWindow and the ClientDlg, but I haven't figured out the right configuration yet.
I'm going to drop it now until the weekend.  Paid work is coming up :)
